frontend/packages/ceph-storage-plugin/src/components/namespace-store/create-ns.tsx

```diff
@@ -0,0 +1,251 @@
+
+const initialState: ProviderDataState = {
+  secretName: '',
+  secretKey: '',
+  accessKey: '',
+  region: AWS_REGIONS[0],
+  target: '',
+  endpoint: '',
+};
+
+const providerDataReducer = (state: ProviderDataState, action: StoreAction) => {
+  const { value } = action;
+  switch (action.type) {
+    case 'setSecretName':
+      return Object.assign({}, state, { secretName: value });
+    case 'setSecretKey':
+      return Object.assign({}, state, { secretKey: value });
+    case 'setAccessKey':
+      return Object.assign({}, state, { accessKey: value });
+    case 'setRegion':
+      return Object.assign({}, state, { region: value });
+    case 'setTarget':
+      return Object.assign({}, state, { target: value });
+    case 'setEndpoint':
+      return Object.assign({}, state, { endpoint: value });
+    default:
+      return initialState;
+  }
+};
+
+const CreateNamespaceStoreForm: React.FC<CreateNamespaceStoreFormProps> = withHandlePromise<
+  CreateNamespaceStoreFormProps & HandlePromiseProps
+>((props) => {
+  const { t } = useTranslation();
+  const [nsName, setNsName] = React.useState('');
+  const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
+  const [providerDataState, providerDataDispatch] = React.useReducer(
+    providerDataReducer,
+    initialState,
+  );
+
+  const handleNsNameTextInputChange = (strVal: string) => {
+    if (strVal.length <= 43) {
+      setNsName(strVal);
+    }
+  };
+
+  const {
+    cancel,
+    className,
+    close,
+    inProgress,
+    errorMessage,
+    handlePromise,
+    isPage,
+    csv,
+    namespace,
+  } = props;
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    /** Create a secret if secret ==='' */
+    let { secretName } = providerDataState;
+    const promises = [];
+    if (!secretName) {
+      secretName = nsName.concat('-secret');
+      const { secretKey, accessKey } = providerDataState;
+      const secretPayload = secretPayloadCreator(
+        provider,
+        namespace,
+        secretName,
+        accessKey,
+        secretKey,
+      );
+      providerDataDispatch({ type: 'setSecretName', value: secretName });
+      promises.push(k8sCreate(SecretModel, secretPayload));
+    }
+    /** Payload for ns */
+    const nsPayload: Payload = {
+      apiVersion: apiVersionForModel(NooBaaNamespaceStoreModel),
+      kind: NooBaaNamespaceStoreModel.kind,
+      metadata: {
+        namespace,
+        name: nsName,
+      },
+      spec: {
+        type: NOOBAA_TYPE_MAP[provider],
+        ssl: false,
+      },
+    };
+    if (externalProviders.includes(provider)) {
+      nsPayload.spec = {
+        ...nsPayload.spec,
+        [PROVIDERS_NOOBAA_MAP[provider]]: {
+          [BUCKET_LABEL_NOOBAA_MAP[provider]]: providerDataState.target,
+          secret: {
+            name: secretName,
+            namespace,
+          },
+        },
+      };
+    }
+    if (provider === BC_PROVIDERS.S3) {
+      // eslint-disable-next-line
```

Comment:
why this ?

Reply:
you can use dot notation like the other ones L154

Reply:
changed it. Sorry lot of code is taken from backing store. I have not reviewed it very well since they both do the same job.